ttnn/cpp/ttnn/operations/ccl/barrier/device/host/barrier_full_worker_grid.cpp

```diff
@@ -0,0 +1,157 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+///
+
+#include "tt_metal/common/core_coord.hpp"
+#include "impl/buffers/buffer.hpp"
+#include "ttnn/operation.hpp"
+#include "ttnn/operations/ccl/ccl_host_types.hpp"
+#include "ttnn/operations/ccl/shared_with_host/hetergeneous_data_structs.hpp"
+#include "ttnn/operations/ccl/ccl_host_datastructures.hpp"
+#include "ttnn/operations/ccl/ccl_common.hpp"
+#include "tt_metal/common/constants.hpp"
+#include "tt_metal/host_api.hpp"
+#include "tt_metal/impl/buffers/circular_buffer_types.hpp"
+
+#include "ttnn/operations/eltwise/binary/common/binary_op_types.hpp"
+#include "ttnn/operations/eltwise/binary/common/binary_op_utils.hpp"
+
+namespace ttnn::ccl::barrier::detail {
+
+
+static std::tuple<KernelHandle,KernelHandle,KernelHandle> schedule_kernel_compile(
+    tt::tt_metal::Program& program,
+    const CoreCoord sender_core,
+    const CoreCoord receiver_core,
+    const CoreCoord sem_init_core
+) {
+    //This just creates a command of what needs to be compiled
+    static std::string const& receiver_code = "ttnn/cpp/ttnn/operations/ccl/barrier/device/kernels/barrier_receiver.cpp";
+    static std::string const& sender_code = "ttnn/cpp/ttnn/operations/ccl/barrier/device/kernels/barrier_sender.cpp";
+    static std::string const& sem_init_code = "ttnn/cpp/ttnn/operations/ccl/barrier/device/kernels/barrier_sem_creator.cpp";
+
+    KernelHandle receiver_kernel_id = tt::tt_metal::CreateKernel(
+        program,
+        receiver_code,
+        receiver_core,
+        tt::tt_metal::EthernetConfig {
+                .compile_args = {}});
+    KernelHandle sender_kernel_id = tt::tt_metal::CreateKernel(
+        program,
+        sender_code,
+        sender_core,
+        tt::tt_metal::EthernetConfig {
+                .compile_args = {}});
+    KernelHandle sem_init_id = tt::tt_metal::CreateKernel(
+        program,
+        sem_init_code,
+        sem_init_core,
+        tt::tt_metal::DataMovementConfig {
+                .compile_args = {}});
+    return {receiver_kernel_id, sender_kernel_id,sem_init_id};
+}
+
+static std::tuple<std::vector<uint32_t>,std::vector<uint32_t>,std::vector<uint32_t>> get_rt_args(
```

Comment:
+ @sminakov-tt 
@jvegaTT is it possible for you to use small_vector or std::array?

Reply:
I was worried that tt::tt_metal::SetRuntimeArgs when I read it expected std::vector but I just noticed that a variant of that function exists using stl::Span which should work with std::array. I am running tests just in case but this change should be ok.

Reply:
It passed the tests